data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/OrderedAsyncExecutor.java

```diff
@@ -42,38 +45,41 @@ public OrderedAsyncExecutor() {
    * @param task the task to offer
    */
   public void offer(Supplier<Future<?>> task) {
-    boolean wasEmpty = queue.isEmpty();
-    queue.offer(task);
+    if (this.isStopped) {
```

Comment:
wanna add a comment here, that if it is stopped, we `return` b/c the queue is empty?

Reply:
sure!